# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/reftests/display-list/1544948-1.html
# Commit: 900ceaf4d39c
# Full Hash: 900ceaf4d39c8968eb4ba8c7769b47064f34ad18
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2019-04-29 09:53:17
# Regressor Bug: 1544948
# File Overlap Count: 1
# Description:
#   Bug 1544948 - Skip merging display lists that we're sure can't have changed. r=miko
#   
#   ComputeRebuildRegion sets ForceDescendIntoIfVisible on all modified frames and their ancestors,
#   so we can use this to detect if a display list might have modified children by looking for this
#   flag on the container item.
# ==============================================================================

diff -r 90c4bb8c0d5c -r 900ceaf4d39c layout/reftests/display-list/1544948-1.html
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/layout/reftests/display-list/1544948-1.html	Mon Apr 29 03:14:49 2019 +0000
@@ -0,0 +1,43 @@
+<html class="reftest-wait">
+<head>
+<style>
+  div {
+    width:10px;
+    height:10px;
+    background-color:green;
+    display: inline-block;
+    position: relative;
+  }
+  #wrapper {
+    opacity: 0.5;
+    width: 200px;
+    height: 200px;
+    background-color: transparent;
+  }
+
+  #first {
+    z-index: 1;
+  }
+</style>
+<script>
+function doTest2() {
+  var elem = document.getElementById("second");
+  elem.style.backgroundColor = "blue";
+}
+
+function doTest() {
+  var elem = document.getElementById("third");
+  elem.parentNode.removeChild(elem);
+  document.documentElement.removeAttribute("class");
+  requestAnimationFrame(doTest2);
+}
+
+window.addEventListener("MozReftestInvalidate", doTest);
+</script>
+</head>
+<body id="body">
+  <div id="wrapper">
+    <div id="first" class="reftest-no-display-list"></div><div id="second"></div><div id="third"></div>
+  </div>
+</body>
+</html>